pywinauto/base_application.py

```diff
@@ -376,21 +592,22 @@ def __getattribute__(self, attr_name):
             # Probably there is no DialogWrapper for another backend
 
             if need_to_resolve:
-                ctrls = self.__resolve_control(self.criteria)
-                return getattr(ctrls[-1], attr_name)
+                ctrl = self.find()
+                return getattr(ctrl, attr_name)
 
         # It is a dialog/control criterion so let getitem
         # deal with it
         return self[attr_name]
 
     def exists(self, timeout=None, retry_interval=None):
         """
-        Check if the window exists, return True if the control exists
+        Wait for the window exists, return True if the control exists.
+
+        :param timeout: maximum length of time to try to find the control.
```

Comment:
`maximum length of time` -> `how much time (in seconds)` (and the same for `not_exists` below)